### ISSUE TITLE
How can I reproduce Emotion-LLaMA_for_EMER.csv?

### ISSUE THREAD

**Issue by 00dbgpdnjs** [ZebangCheng/Emotion-LLaMA#89]
How can I reproduce [Emotion-LLaMA_for_EMER.csv](https://drive.google.com/file/d/1SCrfOtdk7rGlGfp2bQZceZF4ZYpW_pI8/view?usp=drive_link)?
I've attached the log.

[log.txt](https://github.com/user-attachments/files/20684406/log.txt)

Which part should I modify?

**Comment by ZebangCheng** [comment]
**A1:**
I reviewed your log file and it looks correct—no issues there. You're using `EMER_NCE_test.txt`, which is a small test set designed for quick evaluation and only includes a few samples.
To reproduce the full results in `Emotion-LLaMA_for_EMER.csv`, you should use `EMER_NCE.txt`, which contains all evaluation samples. Run inference on this file and then assess the scores.

**Important note:** Due to updates in GPT versions, you may no longer be able to use the same version we used in the official EMER evaluation. To ensure fair and comparable evaluation, I recommend **running both predict results and the baseline** on the same GPT version.

**Pro tip:**
To improve your EMER score and match our paper results, **you should filter out the "Neutral" category** during training. This is because EMER's evaluation set **does not contain Neutral samples**, but the training data does. Removing Neutral samples during training reduces label mismatch and improves reasoning performance.

---

**A2:**
You're right—the file `transcription_en_all.csv` was **generated by us**, not provided by the MER2023 organizers. MER2023 did not release official transcriptions.

We generated the transcriptions using **speech-to-text (STT) tools**, including the official method provided in the MER2023 baseline and models like **Qwen-Audio**.
As a result, there may be slight differences between `transcription_en_all.csv` and `MER2024/label-transcription.csv`.

Let me know if you’d like guidance on aligning the transcriptions or running the full evaluation.

**Comment by 00dbgpdnjs** [comment]
EMER_NCE_test.txt is the NCE-format conversion of Emotion-LLaMA_for_EMER.csv.
It contains the same 100 samples and matching video names as Emotion-LLaMA_for_EMER.csv.

---

In order for the Emotion-LLaMA model to generate the same responses as those in Emotion-LLaMA_for_EMER.csv,
should the model be trained not using Emotion-LLaMA/checkpoints/save_checkpoint/Emotion_LLaMA.pth,
but instead by removing neutral data from MERR_coarse_grained.txt and MERR_fine_grained.txt before training?

**Comment by ZebangCheng** [comment]
yes, you are right. 

> EMER_NCE_test.txt is the NCE-format version of Emotion-LLaMA_for_EMER.csv. Both EMER_NCE_test.txt and Emotion-LLaMA_for_EMER.csv contain the same number of data samples (100) and matching video names.
> 
> So, just to confirm — the model does not use Emotion-LLaMA/checkpoints/save_checkpoint/Emotion_LLaMA.pth, right? Instead, should the model be trained on MERR_coarse_grained.txt and MERR_fine_grained.txt after removing the neutral data, in order to generate the same responses as those in Emotion-LLaMA_for_EMER.csv?

**Comment by 00dbgpdnjs** [comment]
1. I noticed that both the [demo](https://github.com/ZebangCheng/Emotion-LLaMA/blob/main/minigpt4/conversation/conversation.py#L254) and [evaluation](https://github.com/ZebangCheng/Emotion-LLaMA/blob/main/minigpt4/datasets/datasets/first_face.py#L167) use only the first frame of the video for the global encoder. Is that correct?

2. Have you ever tried using a different frame, such as a middle frame, instead of the first frame for the global encoder?

**Comment by 00dbgpdnjs** [comment]
1. When I set `low_resource: False` in demo.yaml, the performance actually dropped. Do you know why that might be?

2. On page 9 of the paper, regarding the combination of encoders (e.g., HuBERT, MAE), were only two encoders used during both training and evaluation? Or were all encoders used during training, and only two selected during testing?

**Comment by ZebangCheng** [comment]
1. I'm not exactly sure why setting `low_resource: False` would lead to a drop in performance. For better evaluation results, I suggest setting `low_resource: True` as shown in `eval_configs/eval_emotion.yaml`, and writing the corresponding evaluation code accordingly.

2. During training, all encoders were used. For evaluation on datasets such as MER2023 and DFEW, the full set of encoders was also employed. However, in the demo deployment, only the Audio Encoder (HuBERT) and the Global Visual Encoder (EVA) were used.

**Comment by 00dbgpdnjs** [comment]
2. I have a question regarding the F1 scores based on the encoder combinations presented on page 9 of your paper.

paper:

     HuBERT + MAE → 0.8800
     HuBERT + VideoMAE → 0.8805
     ...
  
Were these results obtained using the same encoder combinations during both training and evaluation?
Or did you train the model using all encoders and only use the specific combination (e.g., HuBERT + VideoMAE) during evaluation to achieve 0.8805?

Also, I’d like to confirm if your previous response was addressing this specific question.

**Comment by ZebangCheng** [comment]
There are no other differences. I hope you achieve good results and look forward to your feedback.

**Comment by 00dbgpdnjs** [comment]
In the demo code, I noticed it mentions a webcam, but when I run the demo, it only allows video uploads and doesn’t work with a webcam. Is this the intended behavior?

https://github.com/ZebangCheng/Emotion-LLaMA/blob/main/app.py#L598

**Comment by ZebangCheng** [comment]
The demo interface may show a webcam option due to the default Gradio component settings, but in reality, we do not support webcam input at this time.

**Comment by 00dbgpdnjs** [comment]
### about training (total steps)

The paper states a total of 300,000 steps, but:

MERR dataset : 

- step 1

```
max_epoch: 30
iters_per_epoch: 5000
```

- step 2

```
max_epoch: 30
iters_per_epoch: 1000
```


Isn't that 30 × 5000 + 30 × 1000 = 180,000 steps in total?

By any chance, does the total of 300,000 steps mentioned in the paper include training on the DFEW dataset?

Thank you!

**Comment by oAyanameio** [comment]
> I am converting the evaluation code into single-instance inference code.
> 
> This is because the demo shows lower performance compared to the evaluation code.
> 
> Assuming , are there any differences between the demo and the evaluation code besides the following?`row_resource=False`
> 
> * The demo disables both local and temporal features by setting them to 0.

Hello, may I ask if you have successfully converted the evaluation code into single-instance inference code?
Can I take a look at the code?
 Thank you!
